src/discovery/Akka.Discovery.AwsApi/README.md

```diff
@@ -12,37 +12,106 @@ Simply add the tag to the auto-scaling group configuration and ensure the “Tag
+    # to the list of Akka Management port numbers
+    ports = []
+
+    # client may use specified endpoint for example ec2.us-west-1.amazonaws.com
+    # region is automatically extrapolated from the endpoint URL
+    # endpoint = ""
+    
+    # client may use specified region for example us-west-1
+    # endpoints are automatically generated.
+    # NOTE: You can only set either an endpoint OR a region, not both.
+    #       Region will always win if both are declared.
+    # region = ""
+    
+    anonymous-credential-provider {
+        # Fully qualified class name of a class that extends Akka.Discovery.AwsApi.Ec2.Ec2ConfigurationProvider with either 
+        # a no arguments constructor or a single argument constructor that takes an ExtendedActorSystem
+        class = "Akka.Discovery.AwsApi.Ec2.AnonymousEc2CredentialProvider, Akka.Discovery.AwsApi"    
+    }
+    
+    # This configuration provider leverages the EC2 instance metadata service to provide connection
+    # information for the AWS EC2 client.
+    # Please read https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html
+    # on AWS EC2 metadata profile setup.
+    instance-metadata-credential-provider {
+        # Fully qualified class name of a class that extends Akka.Discovery.AwsApi.Ec2.Ec2ConfigurationProvider with either 
+        # a no arguments constructor or a single argument constructor that takes an ExtendedActorSystem
+        class = "Akka.Discovery.AwsApi.Ec2.Ec2InstanceMetadataCredentialProvider, Akka.Discovery.AwsApi"
+        
+        # Name of the Amazon IAM Role to be used as credential provider
+        # If null or entry, the first returned credential is used.
+        role = ""
+    }    
+  }
+}
+```
+
+## Configuration
+### Client configuration
+You can extend the `Amazon.EC2.AmazonEC2Config` class to provide your own configuration
+implementation for the internal EC2 client; the extended class can have either an empty
+constructor or a constructor that takes an ExtendedActorSystem as a parameter. 
+You then provide the fully qualified class name of your implementation in the 
+`akka.discovery.aws-api-ec2-tag-based.client-config` property inside the HOCON configuration.
+
+### Client credentials configuration
+Client credentials are provided by the `Akka.Discovery.AwsApi.Ec2.Ec2CredentialProvider` abstract
+class. There are two implementation provided out of the box, `AnonymousEc2CredentialProvider` and
+`Ec2InstanceMetadataCredentialProvider`.
+
+#### AnonymousEc2CredentialProvider
+`AnonymousEc2CredentialProvider` is a very simple credential provider and will return
+an `AnonymousAWSCredentials`.
+
+#### Ec2InstanceMetadataCredentialProvider
+`Ec2InstanceMetadataCredentialProvider` will try its best to retrieve the correct session
+credential provider using the AWS EC2 instance metadata API. It will return an `AnonymousAWSCredential`
+if it fails to obtain a credential from the metadata API service.
```

Comment:
LGTM

Reply:
We should probably add a link back to the relevant AWS documentation here though

Reply:
Its on the bottom of the readme